client/scala/src/main/scala/ai/verta/client/FormatUtils.scala

```diff
@@ -0,0 +1,11 @@
+package ai.verta.client
+
+import java.net.URLEncoder
+
+/** Containing the helper functions to format the input string */
+object FormatUtils {
```

Comment:
Which then leaves this `utils`, which is bound to become a kitchen sink. Can we do something with `urlEncode` similar to what you did with `jsonFormat`?

Reply:
Hmm I think I can. I just replace your url encode function inside HttpClient (adding the replacement of "+" with "%20"), then split the path string and apply that function to each parts, then join it again. It seems to pass all of my tests (not breaking anything). I've removed FormatUtils entirely. See if that sounds right. Resolved.